### PR TITLE
change: rename locals to debugger (#2503)

### PR DESCRIPTION
**Pull request by hserranome**
## Description

Change proposed in https://github.com/continuedev/continue/issues/2503#issuecomment-2408144695

Renamed "locals" to "debugger"
Renamed LocalsProvider to DebugLocalsProvided for better clarity
Updated descriptions to specify that "local variables" refers to the debugger

## Checklist

- [x] The base branch of this PR is `dev`, rather than `main`
- [x] The relevant docs, if any, have been updated or created

## Screenshots

Screenshot 2024-10-12 144848
image

## Testing

Use the debugger context provider as if it were the locals provided

### CODE REVIEW

File: extensions/vscode/textmate-syntaxes/MagicPython.tmLanguage.json [hunk 1]
```diff
@@ -1569,7 +1569,7 @@
 			"patterns": [
 				{
 					"name": "support.function.builtin.python",
-					"match": "(?x)\n  (?<!\\.) \\b(\n    __import__ | abs | all | any | ascii | bin | callable\n    | chr | compile | copyright | credits | delattr | dir | divmod\n    | enumerate | eval | exec | exit | filter | format | getattr\n    | globals | hasattr | hash | help | hex | id | input\n    | isinstance | issubclass | iter | len | license | locals | map\n    | max | memoryview | min | next | oct | open | ord | pow | print\n    | quit | range | reload | repr | reversed | round\n    | setattr | sorted | sum | vars | zip\n  )\\b\n"
+          "match": "(?x)\n  (?<!\\.) \\b(\n    __import__ | abs | all | any | ascii | bin | callable\n    | chr | compile | copyright | credits | delattr | dir | divmod\n    | enumerate | eval | exec | exit | filter | format | getattr\n    | globals | hasattr | hash | help | hex | id | input\n    | isinstance | issubclass | iter | len | license | debugger | map\n    | max | memoryview | min | next | oct | open | ord | pow | print\n    | quit | range | reload | repr | reversed | round\n    | setattr | sorted | sum | vars | zip\n  )\\b\n"
```
Comment (sestinj) [inline extensions/vscode/textmate-syntaxes/MagicPython.tmLanguage.json]: this shouldn't be changed

File: core/context/providers/LocalsProvider.ts [hunk 2]
```diff
@@ -7,11 +7,12 @@ import {
 } from "../../index.js";
```
Comment (sestinj) [inline core/context/providers/LocalsProvider.ts:7]: can the name of this file also be changed?

**Review by sestinj — CHANGES_REQUESTED** [review]
all looks good, name makes much more sense! just 2 changes

File: extensions/vscode/textmate-syntaxes/MagicPython.tmLanguage.json (hunk 1 above)
Comment (hserranome) [reply extensions/vscode/textmate-syntaxes/MagicPython.tmLanguage.json]: Yeah, got to eager with the find and replace. Fixed.

File: core/context/providers/LocalsProvider.ts (hunk 2 above)
Comment (hserranome) [reply core/context/providers/LocalsProvider.ts:7]: Yep. Done.

File: core/context/providers/DebugLocalsProvider.ts [hunk 3]
```diff
@@ -7,11 +7,11 @@ import {
 } from "../../index.js";
 import { BaseContextProvider } from "../index.js";
 
-class LocalsProvider extends BaseContextProvider {
```
Comment (zzulanas) [inline core/context/providers/DebugLocalsProvider.ts:10]: ❓ Hmmm would it be clearer to name this `DebuggerLocalsProvider`? Not sure if keeping it as "Debug" is clear that this is for debugger and not debug code.

**Review by zzulanas — COMMENTED** [review]
Just putting my two cents in since I commented about the issue haha, haven't contributed (yet!) but was wondering if it's clearer to rename this for future development

**Review by sestinj — APPROVED** [review]

**Review by Thompson1985 — APPROVED** [review]

**Comment by sestinj** [comment]
Hi @hserranome, yesterday we shared [some updates](https://discordapp.com/channels/1108621136150929458/1131313996750917835/1371996789967687783) with our contributors about how we're aiming to improve the contribution process. Part of this included the addition of a [Contributor License Agreement](https://github.com/continuedev/continue/blob/main/docs/docs/CLA.md) (CLA) to protect both contributors and the project. We're reaching out to ask that previous contributors sign it.

Could you please take a moment to sign, or if you have any questions send me a message? (either here or nate@continue.dev would work)

To do so, you just need to **post a comment below with the following text:**<br/>
```
I have read the CLA Document and I hereby sign the CLA
```

❤️ Thank you for the work you've done on Continue, and let me know if you have any suggestions on how we can make the project even better!

**Pull request merged**